### PR TITLE
Add proxyVarsFromSecret: "" option

### PR DESCRIPTION
**Pull request by jan-brychta**
added option to load OAUTH2_PROXY variables directly from a secret

**Reasoning**
ExternalSecrets with GitLab provider makes this chart really hard to use, since you cannot name a GitLab CI/CD variable using "-" such as `cookie-secret`.

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@jan-brychta the `proxyVarsAsSecrets` is not already enough?
Ref: https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/templates/deployment.yaml#L122-L138

**Comment by pierluigilenoci** [comment]
@jan-brychta is OK; I got it.

You could document it better in the PR so that the difference with the other option is apparent to those using the chart. 
In addition, the chart version also needs a bump.

**Comment by jan-brychta** [comment]
Alright, thanks for the heads up @pierluigilenoci

**Comment by jan-brychta** [comment]
@pierluigilenoci @desaintmartin Hey, would you mind sparing a moment to give me a hand with this? I'd really appreciate it!

**Comment by pierluigilenoci** [comment]
@ jan-brychta, I'm not sure what you want from us. Let me explain it again.

What I asked you to integrate into your PR are two points:

* You are introducing a new way to inject secrets besides what already exists into the pod. Your PR is missing any documentation of the new way. I want to ask you to document better the differences between the two methods in the [README.md](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/README.md?plain=1#L177) so that whoever has to choose which method to use has the cognitive tools to make a choice.

* Currently, the chart does not have a test that allows you to verify that the current injection method works. It would be a plus [to have a test](https://github.com/oauth2-proxy/manifests/tree/main/helm/oauth2-proxy/ci) that confirms that both methods work.

Concerning the PR probably also missing a change to the [file that generates the secret](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/templates/secret.yaml).

**Comment by aslafy-z** [comment]
@pierluigilenoci Can you please fix the deployment by moving the `envFrom` outside of `env`? I think the `envFrom` needs to be gated too. Thank you.

**Comment by pierluigilenoci** [comment]
@aslafy-z please feel free to open a PR to address that.

**Comment by pierluigilenoci** [comment]
@jan-brychta, could you please revisit your PR?

File: helm/oauth2-proxy/templates/deployment.yaml [hunk 1]
```diff
@@ -160,6 +160,11 @@ spec:
               name:  {{ template "oauth2-proxy.secretName" . }}
               key: cookie-secret
         {{- end }}
+        envFrom:
+        {{- with .Values.proxyVarsFromSecret }}  
+        - secretRef:
+            name: {{  . }}
+        {{- end }}
```
Comment (aslafy-z) [inline helm/oauth2-proxy/templates/deployment.yaml:167]:
```suggestion
        {{- with .Values.proxyVarsFromSecret }}  
        envFrom:
        - secretRef:
            name: {{  . }}
        {{- end }}
```
To be moved before line 145.

**Comment by aslafy-z** [comment]
Please note that it should be moved out of the env: block (before line 145 for eg.)

**Comment by aslafy-z** [comment]
My comment https://github.com/oauth2-proxy/manifests/pull/141#issuecomment-1969520087 is still to be addressed

**Comment by pierluigilenoci** [comment]
@aslafy-z, you must ask @jan-brychta to address your comments.

**Comment by pierluigilenoci** [comment]
@aslafy-z or @jan-brychta, could you please take a look and try to push forward the PR?

**Comment by aslafy-z** [comment]
This PR has little value to me. However, I opened #196 with the envFrom placement fixed. I closed it, feel free to re-open it if you want to proceed with it and add tests & mention in the readme. If no answer is received from @jan-brychta, I think you can close this PR.

**Comment by pierluigilenoci** [comment]
Closed in favor of #196

**Comment by jan-brychta** [comment]
Appologies @pierluigilenoci, got busy with other project... Thanks for taking over @aslafy-z !

**Pull request closed without merging**